src/modules/mc_hover_thrust_estimator/MulticopterHoverThrustEstimator.cpp

```diff
@@ -91,20 +91,50 @@ void MulticopterHoverThrustEstimator::updateParams()
 void MulticopterHoverThrustEstimator::Run()
 {
 	if (should_exit()) {
-		_vehicle_local_position_setpoint_sub.unregisterCallback();
+		_vehicle_local_position_sub.unregisterCallback();
 		exit_and_cleanup();
 		return;
 	}
 
-	// new local position estimate and setpoint needed every iteration
-	if (!_vehicle_local_pos_sub.updated() || !_vehicle_local_position_setpoint_sub.updated()) {
+	if (_vehicle_land_detected_sub.updated()) {
+		vehicle_land_detected_s vehicle_land_detected{};
+
+		if (_vehicle_land_detected_sub.copy(&vehicle_land_detected)) {
+			_landed = vehicle_land_detected.landed;
+
+			if (_landed) {
+				_in_air = false;
+			}
+		}
+	}
+
+	if (!_vehicle_local_position_sub.updated()) {
+		return;
+	}
+
+	vehicle_local_position_s local_pos{};
+
+	if (_vehicle_local_position_sub.copy(&local_pos)) {
+		// This is only necessary because the landed
+		// flag of the land detector does not guarantee that
+		// the vehicle does not touch the ground anymore
+		// TODO: improve the landed flag
+		if (local_pos.dist_bottom_valid) {
+			if (!_landed && (local_pos.dist_bottom > 1.f)) {
```

Comment:
Would a threshold smaller than 1 meter work or are there secondary benefits with not starting until out of ground effect?

Reply:
I guess a smaller threshold would also work in most cases but I wanted to be a bit more robust against mediocre altitude estimation

Reply:
I removed the `dist_bottom_valid` and added a comment why we don't need it in 653b407